dev_tools/scripts/update_ginkgo_header.sh

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env bash
+shopt -s globstar
+shopt -s extglob
+
+PLACE_HOLDER="#PUBLIC_HEADER_PLACE_HOLDER"
+
+THIS_DIR=$( cd "$( dirname "${BASH_SOURCE[0]}" )" &>/dev/null && pwd )
+
+ROOT_DIR="${THIS_DIR}/../../"
+INCLUDE_DIR="${ROOT_DIR}/include"
+
+cd ${INCLUDE_DIR}
+
+GINKGO_HEADER_FILE="ginkgo/ginkgo.hpp"
+GINKGO_HEADER_TEMPLATE_FILE="${GINKGO_HEADER_FILE}.in"
+
+HEADER_LIST="global_includes.hpp.tmp"
+# Make sure the file does not already exist.
+# If it does, append ".tmp" to the file name and retest it.
+while true; do
+    if [ -f "${HEADER_LIST}" ]; then
+        HEADER_LIST="${HEADER_LIST}.tmp"
```

Comment:
So the file becomes `global_includes.hpp.tmp.tmp`?. Also, is it important to preserve this file? I thought the purpose of this was to collect the list of headers and that's it.

Reply:
You could just clean it if it exists.

Reply:
I wanted to prevent a case where a file `global_includes.hpp.tmp` already existed before running the script. If it exists, the file `global_includes.hpp.tmp.tmp` is tested if it exists, and so on.

This is just to make sure nothing is deleted that already existed.

In the end, the created file is deleted.

Reply:
Well yes, but there are two things I see here:
1. There is no way we will add/commit a file with `.tmp` at the end
2. If the script crashes due to some bug at some point, the user (everyone) will get a huge accumulation of `global_includes.hpp.tmp...`. This is nice for debugging but as a default mode I am not sure. I would rather that nothing is changed at all if a failure occurs, because `make` will (should) report an error on the script's target.

Reply:
Makes sense, I will remove it.